src/v2/Apps/Auction2/Routes/Bid/__tests__/Auction2BidRoute.jest.tsx

```diff
@@ -1,19 +1,249 @@
+jest.mock("v2/Apps/Auction2/Routes/Bid/useSubmitBid", () => ({
+  useSubmitBid: jest.fn(),
+}))
+
+jest.mock("v2/Components/CreditCardInput/CreditCardInputProvider", () => ({
+  CreditCardInputProvider: ({ children }) => children,
+}))
+
+jest.mock("v2/Apps/Auction2/Components/Form/ConditionsOfSaleCheckbox", () => ({
+  ConditionsOfSaleCheckbox: () => null,
+}))
+
+jest.mock(
+  "v2/Apps/Auction2/Routes/Bid/Components/PricingTransparency2",
+  () => ({
+    PricingTransparency2QueryRenderer: () => null,
+  })
+)
+
+jest.mock("v2/Apps/Auction2/Components/Form/AddressFormWithCreditCard", () => ({
+  AddressFormWithCreditCard: () => null,
+}))
+
+jest.mock("v2/Apps/Auction2/Components/Form/ErrorStatus", () => ({
+  ErrorStatus: () => null,
+}))
+
+jest.mock("v2/Apps/Auction2/Routes/Bid/Components/AuctionLotInfo", () => ({
+  AuctionLotInfoFragmentContainer: () => null,
+}))
+
+jest.mock("@artsy/palette", () => ({
+  ...jest.requireActual("@artsy/palette"),
+  ModalDialog: ({ children }) => children,
+}))
+
+jest.mock("formik", () => ({
+  ...jest.requireActual("formik"),
+  Form: ({ children }) => children,
+  Formik: jest.fn(),
+}))
+
 describe("Auction2BidRoute", () => {
-  it("submits bid", () => {})
+  const mockUseAuctionTracking = useAuctionTracking as jest.Mock
+  const mockUseRouter = useRouter as jest.Mock
+  const mockUseSubmitBid = useSubmitBid as jest.Mock
+  const mockFormik = Formik as jest.Mock
+
+  const { getWrapper } = setupTestWrapper<Auction2BidRouteTestQuery>({
+    Component: (props: any) => {
+      return (
+        <Auction2BidRouteFragmentContainer
+          {...props}
+          requiresPaymentInformation
+          requiresCheckbox
+        />
+      )
+    },
+    query: graphql`
+      query Auction2BidRouteTestQuery @relay_test_operation {
+        artwork(id: "foo") {
+          ...Auction2BidRoute_artwork
+        }
+        me {
+          ...Auction2BidRoute_me
+        }
+        sale(id: "foo") {
+          ...Auction2BidRoute_sale
+        }
+      }
+    `,
+  })
+
+  const values = {}
+  const helpers = {}
+
+  const defaultFormikProps = {
+    values,
+    touched: {},
+    errors: {},
+    isSubmitting: false,
+    isValid: true,
+    setFieldError: jest.fn(),
+    setFieldValue: jest.fn(),
+    setFieldTouched: jest.fn(),
+  }
+
+  beforeEach(() => {
+    mockUseAuctionTracking.mockImplementation(() => ({
+      tracking: {
+        bidPageView: jest.fn(),
+      },
+    }))
+
+    mockUseRouter.mockImplementation(() => ({
+      match: {
+        location: {
+          query: {
+            bid: "1000",
+          },
+        },
+        params: {
+          foo: "bar",
+        },
+      },
+    }))
+
+    mockUseSubmitBid.mockImplementation(() => ({
+      submitBid: jest.fn(),
+    }))
+
+    // Hack to get around mocking Formik
+    // @ts-ignore
+    mockFormik.displayName = "Formik"
+    mockFormik.mockImplementation(({ children }) => {
+      return children(defaultFormikProps)
+    })
+  })
```

Comment:
And this is a good way to test Formik behavior without needing to wire up all of the library baggage (which has already been tested)